intelmq/bots/outputs/elasticsearch/output.py

```diff
@@ -74,11 +76,17 @@ def init(self):
         if isinstance(self.flatten_fields, str):
             self.flatten_fields = self.flatten_fields.split(',')
 
-        kwargs = {}
-        if self.http_username and self.http_password:
-            kwargs = {'http_auth': (self.http_username, self.http_password)}
-
-        self.es = Elasticsearch([{'host': self.elastic_host, 'port': self.elastic_port}], **kwargs)
+        self.set_request_parameters()  # Not all parameters set here are used by the ES client
+
+        self.es = Elasticsearch([{'host': self.elastic_host, 'port': self.elastic_port}],
+                                http_auth=self.auth,
+                                use_ssl=self.use_ssl,
+                                verify_certs=self.http_verify_cert,
+                                ca_certs=self.ssl_ca_certificate,
+                                ssl_show_warn=self.ssl_show_warnings,
+                                # client_cert=self.ssl_client_cert,
```

Comment:
What is this comment for?

Reply:
Just suggestions. The ES client supports TLS client auth behind the scenes using those parameter names. It may be that un-commenting those lines and passing the parameters to the bot (ssl_client_cert and key) will work as-is, I just don't have a way to try myself it at the moment. If I get the chance to set up an Elasticsearch node supporting client auth I will revisit it personally, otherwise if someone has a setup already they can un-comment and try it out.